### PR TITLE
Add /nick command for player

### PR DESCRIPTION
**Pull request by hungthezorba**
#55

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -115,9 +116,20 @@ io.on('connection', socket => {
 
   socket.on('message', (content) => {
     io.to(gameID).emit('anon message', {
+      sender: playerNickname,
       content,
       playerID: socket.id
     })
+    if (content.includes("/nick ")) {
+      const tempPlayerNickname = content.substring(5).trim()
+      const regex = new RegExp("^([a-zA-Z0-9_]{1,15})$")
```
Comment (olavfosse) [inline index.js]: We cannot trust the client.

File: index.js [hunk 2]
```diff
@@ -30,6 +30,7 @@ const createGame = _ => ({
 
 io.on('connection', socket => {
   let gameID
+  let playerNickname = 'anon'
```
Comment (olavfosse) [inline index.js]: the username must be read from and written to cookies.

File: index.js [hunk 3]
```diff
@@ -115,9 +116,14 @@ io.on('connection', socket => {
 
   socket.on('message', (content) => {
     io.to(gameID).emit('anon message', {
+      sender: playerNickname,
       content,
       playerID: socket.id
     })
+    if (content.includes("/nick")) {
```
Comment (olavfosse) [inline index.js]: ??????

**Comment by olavfosse** [comment]
Here is an example for how to properly dispatch commands.

https://github.com/olav35/liracer-old/blob/7682f2015664719a613d10d55b85139bf7702dae/front/src/components/Game.js#L92-L114

File: index.js (hunk 1 above)
Comment (hungthezorba) [reply index.js]: Can you explain more the issue ?

Comment (olavfosse) [reply index.js]: Never mind. I thought this was on the client for a second. My bad.

**Comment by hungthezorba** [comment]
> Here is an example for how to properly dispatch commands.
> 
> https://github.com/olav35/liracer-old/blob/7682f2015664719a613d10d55b85139bf7702dae/front/src/components/Game.js#L92-L114

So you want to have a separate event for ("change nick") on client and server ?

**Comment by olavfosse** [comment]
> > Here is an example for how to properly dispatch commands.
> > 
> > https://github.com/olav35/liracer-old/blob/7682f2015664719a613d10d55b85139bf7702dae/front/src/components/Game.js#L92-L114
> 
> So you want to have a separate event for ("change nick") on client and server ?

That is not necesarry. The problem is that merely testing if the chatinput contains /nick is not sufficient. You must test that the first word is /nick.

**Comment by hungthezorba** [comment]
> the username must be read from and written to cookies.

I don't have much experience on processing cookies. Can you check the latest commit if it fit your idea ?

**Comment by olavfosse** [comment]
Functionally it looks good to me, I'll take a closer look at the code later.

Btw I think we should ban users from using liracer as a nick. Also I think the ran commands should only be displayed to the user who ran them.

**Comment by hungthezorba** [comment]
> Btw I think we should ban users from using liracer as a nick. Also I think the ran commands should only be displayed to the user who ran them.

Yeah, the suggestions seem understandable. Btw, do you think the client will hide the ran commands or the server will do that ?

**Comment by olavfosse** [comment]
>Btw, do you think the client will hide the ran commands or the server will do that ?

I'm assuming you mean whether the sending client should hide it. I
don't have a set opinion on that right now.

Med vennlig hilsen / Best regards

Olav Fosse

Happy Hacking!

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by olavfosse** [comment]
Sorry for being slow. I will process this soon.

**Comment by hungthezorba** [comment]
> Sorry for being slow. I will process this soon.

This is an open source, take your time.

**Comment by olavfosse** [comment]
I changed some stuff and I'm merging now.

**Pull request merged**